packages/expo-network/src/Network.ts

```diff
@@ -72,3 +76,66 @@ export async function isAirplaneModeEnabledAsync(): Promise<boolean> {
   }
   return await ExpoNetwork.isAirplaneModeEnabledAsync();
 }
+
+/**
+ * Adds a listener that will fire whenever the network state changes.
+ *
+ * @param listener Callback to execute when the network state changes. The callback is provided with
+ * a single argument that is an object containing information about the network state.
+ *
+ * @example
+ * ```typescript
+ * const subscription = addNetworkStateListener(({ type, isConnected, isInternetReachable }) => {
+ *   console.log(`Network type: ${type}, Connected: ${isConnected}, Internet Reachable: ${isInternetReachable}`);
+ * });
+ * ```
+ *
+ * @returns A subscription object with a remove function to unregister the listener.
+ */
+export function addNetworkStateListener(
+  listener: (event: NetworkStateEvent) => void
+): Subscription {
+  return emitter.addListener<NetworkStateEvent>(onNetworkStateEventName, listener);
+}
+
+/**
+ * Removes the network state listener added by addNetworkStateListener.
+ *
+ * @param subscription The subscription to remove (created by addNetworkStateListener).
+ *
+ * @example
+ * ```typescript
+ * const subscription = addNetworkStateListener(({ isConnected }) => {
+ *   console.log('Network connectivity changed:', isConnected);
+ * });
+ * removeNetworkStateListener(subscription);
+ * ```
+ */
+export function removeNetworkStateListener(subscription: Subscription) {
+  emitter.removeSubscription(subscription);
+}
```

Comment:
Do we need this? `addNetworkStateListener` will return a subscription. We have the hook to take care of this

Reply:
I implemented it because some other modules implement it (e.g. `expo-clipboard`). But the question is whether `removeSubscription` is needed at all, because `addListener` returns the `remove` function.